### PR TITLE
Mock resources

### PR DESCRIPTION
**Pull request by kilchenmann**
This PR contains:

- refactor file names for mocked test data
- export of additional mocked test data: list of resources as ReadResourceSequence; needed for [PR #122 in ui-lib](https://github.com/dasch-swiss/dsp-ui-lib/pull/122)

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
@kilchenmann What is this PR about?

**Comment by tobiasschweizer** [comment]
Would you like to resolve https://github.com/dasch-swiss/knora-api-js-lib/issues/62?

**Comment by kilchenmann** [comment]
> Would you like to resolve #62?

Not primary. I wanted to add a resources-mock file in test/data and saw the different names for the other mock-files. So, the first two commits here were only to make the file names more consistent.

> What is this PR about?

I have to mock a ReadResourceSequence to use for the tests in ui-lib. I will add a description to the PR soon. 
Let me know if this is the wrong way. We talked about it in the scrum meeting today.

**Comment by tobiasschweizer** [comment]
> > Would you like to resolve #62?
> 
> Not primary. I wanted to add a resources-mock file in test/data and saw the different names for the other mock-files. So, the first two commits here were only to make the mock data file names more consistent.
> 
> > What is this PR about?
> 
> I have to mock a ReadResourceSequence to use for the tests in ui-lib. I will add a description to the PR soon.
> Let me know if this is the wrong way. We talked about it in the scrum meeting today.

I understand. Sorry, I didn't get it in the first place. Maybe you could just make a mock that returns a `ReadResourceSequence` and internally calls `MockResourcegetTestthing()`. Just make sure that the sequence contains **different** instances of testthing (objects are reference types). Just call `MockResource.getTestthing()` as many times as there are elements in the mocked `ReadResourceSequence `.

Otherwise I can help you with that next week.

File: test/data/api/v2/mock-resources.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { ListNodeV2Cache } from "../../../../src/cache/ListNodeV2Cache";
+import { OntologyCache } from "../../../../src/cache/ontology-cache/OntologyCache";
+import { KnoraApiConfig } from "../../../../src/knora-api-config";
+import { ReadResourceSequence } from "../../../../src/models/v2/resources/read/read-resource-sequence";
+import { ResourcesConversionUtil } from "../../../../src/models/v2/resources/ResourcesConversionUtil";
+import testthings from "../v2/resources/things-expanded.json";
+import { MockList } from "./mock-list";
+import { MockOntology } from "./mock-ontology";
+
+export namespace MockResources {
```
Comment (tobiasschweizer) [inline test/data/api/v2/mock-resources.ts:15]: I think you could just add a new method to the existing `MockResource`:

```ts
    export const getTesthings = (length = 25): Observable<ReadResourceSequence> => {

        const resources: Array<Observable<ReadResource>> = new Array(length).fill(0).map(
            () => {
                return getTestthing();
            }
        );

        return forkJoin(resources).pipe(
            map(
                (res: ReadResource[]) => {
                    return new ReadResourceSequence(res);
                }
            )
        );

    };
```

You could even add an arg to set `mayHaveMoreResults` on `ReadResourceSequence` to make it more flexible.

Comment (tobiasschweizer) [reply test/data/api/v2/mock-resources.ts:15]: this is some `RxJS` magic: https://www.learnrxjs.io/learn-rxjs/operators/combination/forkjoin

Comment (kilchenmann) [reply test/data/api/v2/mock-resources.ts]: Resolved in 723dfbb and 63d2a25

File: test/data/api/v2/mock-resource.ts [hunk 2]
```diff
@@ -59,4 +59,22 @@ export namespace MockResource {
 
     };
 
+    export const getTesthings = (length = 25): Observable<ReadResourceSequence> => {
```
Comment (tobiasschweizer) [inline test/data/api/v2/mock-resource.ts:62]: If you want, you could pass a Boolean arg that would set `mayHaveMoreResults`. Maybe that could be useful for testing the search result component?

File: test/data/api/v2/mock-resources.ts (hunk 1 above)
Comment (kilchenmann) [reply test/data/api/v2/mock-resources.ts]: Nice solution btw

File: test/data/api/v2/mock-resource.ts (hunk 2 above)
Comment (kilchenmann) [reply test/data/api/v2/mock-resource.ts:62]: Yes, why not?

File: test/data/api/v2/mock-resource.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export namespace MockResource {
 
     };
 
-    export const getTesthings = (length = 25): Observable<ReadResourceSequence> => {
+    export const getTesthings = (length = 25, mayHaveMoreResults = false): Observable<ReadResourceSequence> => {
```
Comment (kilchenmann) [inline test/data/api/v2/mock-resource.ts:62]: @tobiasschweizer That should work like this, right?

File: test/data/api/v2/mock-resource.ts (hunk 2 above)
Comment (kilchenmann) [reply test/data/api/v2/mock-resource.ts]: Done in 
af0d0d6

File: test/data/api/v2/mock-resource.ts [hunk 4]
```diff
@@ -59,4 +59,22 @@ export namespace MockResource {
...
+        const resources: Array<Observable<ReadResource>> = new Array(length).fill(0).map(
+            () => {
+                return getTestthing();
+            }
+        );
+
+        return forkJoin(resources).pipe(
+            map(
+                (res: ReadResource[]) => {
+                    return new ReadResourceSequence(res, mayHaveMoreResults);
```
Comment (kilchenmann) [inline test/data/api/v2/mock-resource.ts:73]: @tobiasschweizer That should work like this, right?

Comment (tobiasschweizer) [reply test/data/api/v2/mock-resource.ts:73]: Yes, I think so.

**Comment by kilchenmann** [comment]
> I think this PR is good to go now.

Thanks

**Pull request merged**